go/vt/vtgate/semantics/binder.go

```diff
@@ -258,7 +258,7 @@ func (b *binder) resolveColumn(colName *sqlparser.ColName, current *scope, allow
 		}
 		current = current.parent
 	}
-	return dependency{}, NewError(ColumnNotFound, colName)
+	return dependency{}, &ColumnNotFoundError{Column: sqlparser.String(colName)}
```

Comment:
I think we also want the table here as well and ideally keyspace too? Without the table, the error message is not that useful for a caller. In general, I'd suggest we do that in other places where we know that kind of information as well.

Reply:
done

Reply:
not done. Only formalized the column, but did not add a table.

Reply:
@systay what is the correct way of getting the identity of the table here?

Reply:
addressed by @systay in https://github.com/vitessio/vitess/pull/12572/commits/b0d1199044a40c248fc51c434be006974a950f01